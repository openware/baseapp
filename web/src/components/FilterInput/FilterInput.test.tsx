import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { SinonSpy, spy } from 'sinon';
import { FilterInput, FilterInputProps } from './';

const data = [
    {
        active: true,
        address: '0xdeadbeef',
        cryptoAmount: 10,
        cryptoCode: 'BTC',
        fiatAmount: 100,
        fiatCode: 'USD',
        lockedAmount: 0,
    },
    {
        active: false,
        address: '0xdvrervef',
        cryptoAmount: 10,
        cryptoCode: 'ETH',
        fiatAmount: 100,
        fiatCode: 'USD',
        lockedAmount: 0,
    },
];

const defaults: FilterInputProps = {
    data,
    filter: (item: (typeof data)[0], term) => String(item.cryptoCode).toLowerCase().indexOf(term.toLowerCase()) !== -1,
    onFilter: spy(),
};

const renderComponent = (props: Partial<FilterInputProps> = {}) =>
    render(<FilterInput {...{ ...defaults, ...props }} />);

describe('FilterInput', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = renderComponent();
    });

    it('should render', () => {
        expect(renderComponent().container).toMatchSnapshot();
    });

    it('should render 1 input tag', () => {
        expect(screen.getByRole('textbox')).toBeInTheDocument();
    });

    it('should have correct className', () => {
        expect(wrapper.container.querySelector('.cr-search')).toBeInTheDocument();
    });

    it('should render correct list from data', () => {
        fireEvent.change(screen.getByRole('textbox'), { target: { value: 'b' } });

        expect((defaults.onFilter as SinonSpy).calledOnceWith([defaults.data[0]])).toBeTruthy();
    });
});
