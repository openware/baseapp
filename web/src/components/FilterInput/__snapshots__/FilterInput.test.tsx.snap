// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput should render 1`] = `
<div>
  <div
    class="cr-search"
  >
    <span
      class="cr-search__icon"
    >
      <img
        alt="search icon"
      />
    </span>
    <input
      class="cr-search__input"
      placeholder="Search"
      type="text"
      value=""
    />
    <span
      class="cr-search__cancel"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.5 0C11.6475 0 15 3.3525 15 7.5C15 11.6475 11.6475 15 7.5 15C3.3525 15 0 11.6475 0 7.5C0 3.3525 3.3525 0 7.5 0ZM10.1925 3.75L7.5 6.4425L4.8075 3.75L3.75 4.8075L6.4425 7.5L3.75 10.1925L4.8075 11.25L7.5 8.5575L10.1925 11.25L11.25 10.1925L8.5575 7.5L11.25 4.8075L10.1925 3.75Z"
          fill="var(--icons)"
        />
      </svg>
    </span>
  </div>
</div>
`;
